packages/tangle/messagefactory.go

```diff
@@ -129,6 +129,78 @@ func (f *MessageFactory) IssuePayload(p payload.Payload, t ...*Tangle) (*Message
 	return msg, nil
 }
 
+// IssuePayloadWithDelay creates a new message including sequence number and tip selection and returns it.
+// It also triggers the MessageConstructed event once it's done, which is for example used by the plugins to listen for
+// messages that shall be attached to the tangle.
+// It is the modification of IssuePayload method that wait specified time delay after message creation and before issuance
+func (f *MessageFactory) IssuePayloadWithDelay(p payload.Payload, delay time.Duration, t ...*Tangle) (*Message, error) {
+	payloadLen := len(p.Bytes())
+	if payloadLen > payload.MaxSize {
+		err := fmt.Errorf("maximum payload size of %d bytes exceeded", payloadLen)
+		f.Events.Error.Trigger(err)
+		return nil, err
+	}
+
+	f.issuanceMutex.Lock()
+	defer f.issuanceMutex.Unlock()
+	sequenceNumber, err := f.sequence.Next()
+	if err != nil {
+		err = xerrors.Errorf("could not create sequence number: %w", err)
+		f.Events.Error.Trigger(err)
+		return nil, err
+	}
+
+	strongParents, weakParents, err := f.selector.Tips(p, 2, 2)
+	if err != nil {
+		err = xerrors.Errorf("tips could not be selected: %w", err)
+		f.Events.Error.Trigger(err)
+		return nil, err
+	}
+
+	issuingTime := clock.SyncedTime()
```

Comment:
Isn't `issuingTime` now + delay?

Reply:
No, purpose of this is to allow us to issue a transaction with incorrect timestamp, so we do everything as usual but wait the delay before issuing. It's just for testing.